### DISCUSSION TITLE
Linking SQL Server 2016 to a github repository

### DISCUSSION THREAD

**Discussion by Sessagesimale** [community/community#134695] [New to GitHub]
Hi everyone, never used GitHub before.

i was wondering if there's a way to connect my 2016  Microsoft SQL Server DB to a repository. My goal is to track all changes made to the DB and the objects in it.

I was trying this https://docs.devart.com/source-control/linking-to-source-control/linking-db-to-git-repository-in-github.html and i got stuck at 3.1, Apparently i can't install Team Foundation Server 2013&2015 MSSCCI which apparently is mandatory.

Any ideas?

**Comment by GabrieleTronchin** [comment]
Hi @Sessagesimale ,

Interesting post! I hadn't considered putting a database under source control before; typically, you only store some SQL scripts.

By the way, it seems like you need to install [dbForge](https://www.devart.com/dbforge/sql/source-control/). 
Here's the [installation guide](https://docs.devart.com/source-control/getting-started/installing.html).

Additionally, using TFS does not appear to be mandatory, according to the [requirements](https://docs.devart.com/source-control/getting-started/requirements.html) for the package.

dbForge is an add-on for SQL Management Studio, but you might also need to install it on the server where SQL Server is running.

However, I believe your question isn't directly related to GitHub, but rather to the specific use of this Devart tool. Therefore, I suggest reaching out to the company that provides the tool for more detailed information.

**Comment by geniuszly** [comment]
To link your SQL Server 2016 database to a GitHub repository and track changes, you have a few options even if you're having trouble with the Team Foundation Server (TFS) MSSCCI provider. 
1. SQL Server Data Tools (SSDT) is an extension for Visual Studio that allows you to manage your SQL Server database schema. With SSDT, you can create a database project, which can then be linked to a GitHub repository. Open Visual Studio, and create a new SQL Server Database Project. Import your existing database schema into the project. Use GitHub within Visual Studio or command-line Git to link the project to a repository. You'll be able to commit and track changes to the database schema. Any updates to the schema can be done through the project, and then committed to GitHub, ensuring version control over the database objects.
2. Redgate SQL Source Control: Redgate offers a tool called SQL Source Control, which integrates directly with SQL Server Management Studio (SSMS). This tool can link your database directly to a GitHub repository without needing TFS or SSDT. After installation, you can right-click your database in SSMS and choose to link it to a source control system like GitHub.
3. Devart dbForge Source Control: Devart dbForge Source Control is another tool that allows you to link your SQL Server database to GitHub. It's similar to Redgate's tool and also integrates with SSMS.
4. If you don't want to use a specific tool, you can manually script your database objects and check them into a GitHub repository. This method requires discipline but gives you full control. Regularly generate scripts for database objects like tables, views, stored procedures, etc. Use Git to commit these scripts to a GitHub repository.
5. If you specifically need the TFS MSSCCI provider, ensure that your Visual Studio version is compatible. You can try using an older version of Visual Studio or exploring forums for specific installation issues. Alternatively, if TFS is not mandatory, using a Git-only solution like the ones mentioned above would simplify your workflow.
If you're stuck at a specific step in the Devart documentation, you might find it easier to use an alternative tool like Redgate SQL Source Control or dbForge. These tools are designed to integrate seamlessly with GitHub and SQL Server, allowing you to track changes to your database schema effectively without needing TFS MSSCCI.
